### PR TITLE
Add list of design patterns to README.rst

### PR DESCRIPTION
**Pull request by zancas**
Fixes: #501

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -55,6 +55,76 @@ Usage
...
+#. public-nonpublic pairs
+    e.g. 'create' and '_create' XXX add content here.
+#. Handle known issues in the device REST server.
+    The SDK intends to provide
+    a rational interface to consumers that does the right thing.  This means
+    that one case where it does NOT simply do the stupid thing is when it
+    handles a known idiosyncrasy in the device REST server.  For example, some?
+    resources ignore 'disable' and 'enable' configuration options when they are
+    set to 'False', rather than force a consumer to learn about this quirk in
+    the server, then SDK guesses that '"disable": False' means '"enable": True'
```
Comment (caphrim007) [inline README.rst]: "the" instead of "then". "the SDK guesses"

File: README.rst [hunk 2]
```diff
@@ -55,6 +55,76 @@ Usage
...
+
+#. public-nonpublic pairs
+    e.g. 'create' and '_create' XXX add content here.
+#. Handle known issues in the device REST server.
+    The SDK intends to provide
+    a rational interface to consumers that does the right thing.  This means
+    that one case where it does NOT simply do the stupid thing is when it
+    handles a known idiosyncrasy in the device REST server.  For example, some?
+    resources ignore 'disable' and 'enable' configuration options when they are
+    set to 'False', rather than force a consumer to learn about this quirk in
```
Comment (caphrim007) [inline README.rst]: I'd end the sentence after False and just start a new one at "Rather"

**Comment by caphrim007** [comment]
curious what your thoughts on what tool (if not the SDK) would be used for smoothing the user experience. For example, the verbosity of the OpenStack client libraries has resulted in the shade library https://pypi.python.org/pypi/shade

**Comment by zancas** [comment]
I think that 'smoothing' needs to be decomposed into distinct subtasks...   some examples:
- contextmanager-based transactions for specific configurations
- reasonable defaults for different use-cases
- convenience interpreters that interpret and present relevant configuration information

So "clustering" is an example of code that I think has a lot of higher-level logic cleanly separated from the rest of the SDK.   In the near future I think we need to rename the 'bigip' directory/subpackage in a way that reflects that it's really the "core" of the SDK, and then we can add sister subpackages/directories like clustering...

So, in short I think that the SDK should grow higher level logic...   in cleanly separated namespaces.

**Comment by zancas** [comment]
We as a group need to figure out how we want to rename the `bigip` directory.
